codegen/golang.go

```diff
@@ -81,23 +86,37 @@ func WithUnwrapFieldsStructs(fields []string) TableOptions {
 	}
 }
 
-// Unwrap all fields that are embedded structs (1 level deep only)
+// WithUnwrapAllEmbeddedStructs unwraps all fields that are embedded structs (1 level deep only)
 func WithUnwrapAllEmbeddedStructs() TableOptions {
 	return func(t *TableDefinition) {
 		t.unwrapAllEmbeddedStructFields = true
 	}
 }
 
-func DefaultTransformer(field reflect.StructField) string {
+// WithLogger
+func WithLogger(logger zerolog.Logger) TableOptions {
+	return func(t *TableDefinition) {
+		t.logger = logger
+	}
+}
+
+// WithValueTypeTransformer sets a function that can override the schema type for specific fields. Return `schema.TypeInvalid` to fall back to default behavior.
```

Comment:
I think this comment is now incorrect: does it still fall back to default behavior if you return `schema.TypeInvalid`?

Reply:
Yes, it fallbacks, so should state so

Reply:
I think yes. as @candiduslynx requested it. I added the fallback behaviour back.

Reply:
Oh sorry I see, it does yeah.

Reply:
I think we can ad basically you can have either fallback or let's say you want to use `Default` but in some different order 🤷  I would keep both options in that case (even though it's additional external API).